src/stages/stage3_events.c

```diff
@@ -165,30 +162,24 @@ TASK(burst_swirls, { int count; int interval; ProjPrototype *shot_proj; }) {
 // side swirls
 // typically move across a stage in a drive-by fashion
 
-TASK(side_swirl_move, { BoxedEnemy e; cmplx p0; cmplx p1; cmplx p2; } ) {
+TASK(side_swirl_move, { BoxedEnemy e; cmplx vel; cmplx accel; cmplx accel_retention; } ) {
 	Enemy *e = TASK_BIND(ARGS.e);
-	cmplx p0 = ARGS.p0;
-	cmplx p1 = ARGS.p1;
-	cmplx p2 = ARGS.p2;
-
-	for (;;) {
-		// this "move" moves in a kind of "S" shape, horizontally across the screen
-		// TODO: replicate this w/o using `for (;;)`
-		e->move = move_linear(p0 + p1 * creal(p2));
-		p2 = creal(p2) * cimag(p2) + I * cimag(p2);
-		YIELD;
-	}
+	cmplx impulse_vel = ARGS.accel * creal(ARGS.accel_retention);
+	cmplx vel = ARGS.vel;
+	real retention = cimag(ARGS.accel_retention);
+
+	e->move = move_asymptotic(impulse_vel, vel, retention);
 }
 
-TASK(side_swirl, { cmplx pos; cmplx p0; cmplx p1; cmplx p2; }) {
+TASK(side_swirl, { cmplx pos; cmplx vel; cmplx accel; cmplx accel_retention; }) {
```

Comment:
`accel_retention` shouldn't be a complex value - make it a real, and pass only the "imaginary" component of the original mystery parameter. The "real" component should be just pre-multiplied into `accel`.

Example:
Instead of this:
```c
.accel = -1*I,
.accel_retention = 25+0.95*I
```
you would write:
```c
.accel = -1*I * 25,  // or just -25*I
.accel_retention = 0.95
```
when invoking the swirl task.

`accel` is also not a very good name for this. You can think of the motion described by `move_asymptotic(vel0, vel1, retention)` as a gradual transition from the initial velocity `vel0` (which is what we unfortunately misnamed as "accel" here), into the final velocity `vel1`. Retention can be thought of as a percentage of how much of the original velocity is preserved at each time step (or its exponential decay rate).

Finally, you can dodge the whole naming issue by simply taking the whole `MoveParams` struct as a task argument (by value), and just do this in the timeline:
```c
.move = move_asymptotic(-25*I, 5, 0.95),
```
and this in the task code:
```c
e->move = ARGS.move;
```
It also makes sense because you can make it a simple `move_linear()` for cases where the swirls just move in a straight line at the sides of the screen.